libs/concurrency/include/hpx/concurrency/cache_line_data.hpp

```diff
@@ -40,6 +41,13 @@ namespace hpx {
                            (data_size % threads::get_cache_line_size())) %
                     threads::get_cache_line_size();
             }
+
+            template <typename Data>
+            struct needs_padding
+              : std::integral_constant<bool,
+                    detail::get_cache_line_padding_size(sizeof(Data)) != 0>
+            {
+            };
```

Comment:
Just curious, where do we use this with 0-sized data?

Reply:
It's any data that has a size evenly divisible by the cache line size. I ran into this while working on the new scheduler. MSVC emits a warning for zero-sized arrays.

Reply:
Ah right, makes sense. Thanks!